tests/functional/modules/test_zos_copy_func.py

```diff
@@ -468,6 +468,56 @@ def test_copy_dir_to_existing_uss_dir_not_forced(ansible_zos_module):
         hosts.all.file(path=dest_dir, state="absent")
 
 
+@pytest.mark.uss
+@pytest.mark.parametrize("src", [
+    dict(is_remote=True),
+    dict(is_remote=False),
+])
+def test_copy_subdirs_folders_and_validate_recursive_encoding(ansible_zos_module, src):
+    hosts = ansible_zos_module
+    dest_path = "/tmp/test/"
+    text_outer_file = "Hi I am point A"
+    text_inner_file = "Hi I am point B"
+    src_path = "/tmp/level_1/"
+    outer_file = "/tmp/level_1/text_A.txt"
+    inner_src_path = "/tmp/level_1/level_2/"
+    inner_file = "/tmp/level_1/level_2/text_B.txt"
+
+    try:
+        hosts.all.file(path=inner_src_path, state="directory")
+        hosts.all.file(path=inner_file, state = "touch")
+        hosts.all.file(path=outer_file, state = "touch")
+        hosts.all.shell(cmd="echo '{0}' > '{1}'".format(text_outer_file, outer_file))
+        hosts.all.file(path=dest_path, state="directory")
+        hosts.all.shell(cmd="echo '{0}' > '{1}'".format(text_inner_file, inner_file))
+
+        copy_res = hosts.all.zos_copy(content=src_path, dest=dest_path, encoding={"from": "ISO8859-1", "to": "IBM-1047"}, remote_src=src["is_remote"])
+
+        stat_res = hosts.all.stat(path=src_path)
+        for result in copy_res.contacted.values():
+            assert result.get("msg") is None
+            assert result.get("changed") is True
+
+        for st in stat_res.contacted.values():
+            assert st.get("stat").get("exists") is True
+
+        inner_file_text_aft_encoding = hosts.all.shell(cmd="cat {0}".format(inner_file))
+        outer_file_text_aft_encoding = hosts.all.shell(cmd="cat {0}".format(outer_file))
+        for text in outer_file_text_aft_encoding.contacted.values():
+            text_outer = text.get("stdout")
+        for text in inner_file_text_aft_encoding.contacted.values():
+            text_inner = text.get("stdout")
+
+        assert text_inner == text_inner_file
+        assert text_outer == text_outer_file
+    finally:
+        hosts.all.file(path=inner_file, state="absent")
```

Comment:
Should be enough to remove the hierarchy only with these two right? 

hosts.all.file(path=src_path, state="absent") // "/tmp/level_1/"
hosts.all.file(path=dest_path, state="absent") // "/tmp/test/"